### PR TITLE
Find conda debug issue by passing debug flags into make CMake presets

### LINKED ISSUE #32786: Linux: Conda overwrites CFLAGS with -O2 in Debug Mode
**Describe the bug**
Conda completely ignores the intension to build with minimal optimisations, instead appending it's own CFLAGS over the top as per [this list](https://docs.conda.io/projects/conda-build/en/latest/resources/compiler-tools.html#customizing-the-compilers)

**To Reproduce**
- Activate the `mantid-developer` conda env
- `echo $CFLAGS` notice `-O2` is in the list
- rm the CMakeCache.txt in your build dir
- Configure CMake with `-DCMAKE_BUILD_TYPE=Debug`
- Grep the CMakeCache file, notice `-O2` is littered around

**Expected behavior**
According to [this page - Note on the Compiler Packages](https://www.prouvost.dev/post/2021/c-development-with-conda/) we either use system packages, accept having optimisation on in debug builds, or using a toolchain to manually specify our CFLAGS

### PR DESCRIPTION
**Pull request by StephenSmith25**
**Description of work.**
Conda completely ignores the intension to build with minimal optimisations, instead appending it's own CXXFLAGS. These CXX flags are put into the CMAKE_CXX_FLAGS variable. The debug flags are then appended on, which is typically just appending a "-g" to include the debug symbols. Because the default turns -O2 on we need to overwrite this in the debug flags by passing in -O0. 

I feel this is a better approach than putting compiler specific handling into our CMake files. 

**To test:**

1. cmake .. --preset=linux -DCMAKE_VERBOSE_MAKEFILE=ON
2. ninja -v
3. You could also test debugging some stuff. There shouldn't be any missing or optimised out code.

You should be able to now view the exact compile commands. You'll notice a -O2 followed by a -O0, which overrides the previous setting. 

Fixes #32786. 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by DavidFair — CHANGES_REQUESTED** [review]

File: CMakePresets.json [hunk 1]
```diff
@@ -45,7 +47,8 @@
                 "CONDA_ENV": true,
                 "CMAKE_FIND_USE_SYSTEM_ENVIRONMENT_PATH": "FALSE",
                 "CMAKE_FIND_USE_CMAKE_ENVIRONMENT_PATH": "FALSE",
-                "CMAKE_BUILD_TYPE": "Debug"
+                "CMAKE_BUILD_TYPE": "Debug",
```
Comment (DavidFair) [inline CMakePresets.json:50]: This duplicated field causes JSON parsing to fail

**Comment by DavidFair** [comment]
@StephenSmith25 I hope you don't mind, I pushed the missing comma/duplicated JSON line fix to your branch?
I'm just waiting for the build to go through locally...

**Review by DavidFair — APPROVED** [review]
This works correctly, I could step through a KernelTest target with gdb and no symbols / steps were optimised away. Thanks for fixing that :+1:

**Comment by StephenSmith25** [comment]
Thank you for pushing the fix!

**Pull request merged**